src/frontend/src/utils/i18n.ts

```diff
@@ -0,0 +1,62 @@
+/** Internationalization (i18n) and copy support for Internet Identity.
+ *
+ * This module provides helper for loading multi-language copy definitions and updating the lit templates
+ * depending on the user-selected language dynamically. */
+
+import { TemplateResult, html } from "lit-html";
+import { asyncReplace } from "lit-html/directives/async-replace.js";
+import { Chan } from "./utils";
+
+// The type of raw copy, i.e. { lang: { some_key_title: "Welcome!" } }
+type StringCopy<Keys extends string, Lang extends string> = {
+  [key in Lang]: { [key in Keys]: string };
+};
+
+// The dynamic templates, i.e. { some_key_title: <language-dependent template> }
+type DynamicCopy<Keys extends string> = { [key in Keys]: TemplateResult };
+
+/// A class that converts a string copy definition into dynamic lit templates dependent on the
+// currently selected language.
+export class I18n<Lang extends string> {
+  // The 'Chan' used to propagate language updates
+  private chan: Chan<Lang>;
```

Comment:
not related to the PR question: JavaScript newly I think supports `#` to identify `private` member. Do you know if TypeScript converts them?

`private chan`-> bundle `#chan`?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields

Reply:
Didn't know that was a thing, thanks!

And yes, looks like it's supported: https://www.typescriptlang.org/play?ts=3.8.3#example/private-class-fields